scripts/zones/Empyreal_Paradox/bcnms/Apocalypse_Nigh.lua

```diff
@@ -0,0 +1,47 @@
+-----------------------------------
+-- Area: Empyreal_Paradox
+-- Name: AN
+-- instance 1 Promathia !pos -520 -119 524
+-- instance 2 Promathia !pos 521 -0.500 517
+-- instance 3 Promathia !pos -519 120 -520
+-----------------------------------
+local ID = require("scripts/zones/Empyreal_Paradox/IDs")
+require("scripts/globals/battlefield")
+require("scripts/globals/keyitems")
+require("scripts/globals/missions")
+require("scripts/globals/titles")
+require("scripts/globals/quests")
+-----------------------------------
+
+function onBattlefieldTick(battlefield, tick)
+    tpz.battlefield.onBattlefieldTick(battlefield, tick)
+end
+
+function onBattlefieldRegister(player, battlefield)
+
+end
+
+function onBattlefieldEnter(player, battlefield)
+end
+
+function onBattlefieldDestroy(battlefield)
+end
+
+function onBattlefieldLeave(player, battlefield, leavecode)
+    if leavecode == tpz.battlefield.leaveCode.WON then
+        local name, clearTime, partySize = battlefield:getRecord()
+        player:startEvent(7)
+    elseif leavecode == tpz.battlefield.leaveCode.LOST then
+        player:startEvent(32002)
+    end
+end
+
+function onEventUpdate(player, csid, option)
+end
+
+function onEventFinish(player, csid, option)
+    if csid == 7 then
```

Comment:
this will set the charVar on everyone who participated in the battle.  we only want to set it on people who are currently on this quest.

Reply:
For these two, thinking this would work? 
 onEventFinish(player, csid, option)
    if csid == 32001 and player:getCharVar('ApocalypseNigh') == 4 then
      player:setPos(-.0745,-10,-465.1132,63,33);
      player:setCharVar('ApocalypseNigh',5);
    else
      player:setPos(-.0745,-10,-465.1132,63,33);